weekly-scan/scheduler.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: "v1"
+kind: "Template"
+metadata:
+  name: weekly-scan-scheduler-template
+labels:
+  type: "scheduler"
+  app: "weekly-scan"
+objects:
+  - kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: weekly-scan-scheduler-script
+    data:
+      script: |-
+        GOT_ANY='0'
+        OS_TOKEN=`cat /tmp/jenkins-secret/token`
+        oc login ${OS_MASTER} --insecure-skip-tls-verify=true --token=${OS_TOKEN};
+        for item in `oc get bc -o name --selector app=weekly-scan --selector type=pipeline`; do
+            GOT_ANY='1'
+            oc start-build ${item};
+            sleep ${BATCH_DELAY}
+        done
+        if [ ${GOT_ANY} == '0' ]; then
+            echo "No projects to trigger"
+        fi
+  - apiVersion: batch/v1beta1
+    kind: CronJob
+    metadata:
+      name: weekly-scan-scheduler
+    spec:
+      schedule: "*/1 * * * *"
+      jobTemplate:
+        spec:
+          template:
+            metadata:
+            spec:
+              containers:
+                - name: weekly-scan-scheduler
+                  image: ${JENKINS_SLAVE_IMAGE_NAME}
+                  command: ["/bin/sh",  "/tmp/entry/script"]
+                  volumeMounts:
+                    - name: weeklyscanscript
+                      mountPath: /tmp/entry
+                      readOnly: true
+                    - name: jenkins-secret
+                      mountPath: /tmp/jenkins-secret
+                      readOnly: true
+                  env:
+                    - name: OS_MASTER
+                      valueFrom:
+                        configMapKeyRef:
+                          name: os-master-config
+                          key: url
+              volumes:
+                - name: weeklyscanscript
+                  configMap:
+                    name: weekly-scan-scheduler-script
+                - name: jenkins-secret
+                  secret:
+                    secretName: ${JENKINS_SECRET_NAME}
+              restartPolicy: OnFailure
```

Comment:
`restartPolicy` question is still open. Copying over here:

> What does this restartPolicy exactly do? If cron failed for some reason, it would restart it? What if there's some bug and we end up restarting it each time? Would that send weekly scan emails in a loop to the users?

Reply:
Well, it could happen, although chances of this job failing are pretty slip, unless there is failure in the oc commands. Anycase, if you guys want it removed, i can.